ivadomed/loader/loader.py

```diff
@@ -504,7 +538,7 @@ def __init__(self, root_dir, subject_lst, target_suffix, contrast_lst, contrast_
 class BidsDataset(MRI2DSegmentationDataset):
```

Comment:
Here I was thinking to reduce the number of input parameters by passing directly the `context` variable. The instantiation of these classes is [massive](https://github.com/neuropoly/ivado-medical-imaging/pull/240/files#diff-edd3e7e75e956aeb056f2c88401ec9eaR520) . @charleygros do you see any down sides to this approach?

Reply:
I felt the same while doing the refactoring of `main.py`.
At the same time, I do not want to fall in the other extrem, eg `loader(context)` --> with `context` being an obscure dictionary with all the parameters --> makes it "back-box".
What I am suggesting for the `main.py` (open to discussion) would be a middle ground between the two: params grouped by "application". For instance, I have:
```
def train(model_params, dataset_train, dataset_val, training_params, log_directory, cuda_available=True, metric_fns=None, debugging=False):
```
Where `model_params` and `training_params` are dictionary containing parameters related to model and training_strategy, resp.

Reply:
Okay great! I won't modify it since you already made the changes in the main refactor PR :)!

Reply:
Another potential solution to keep the code clean and avoid repeating arguments is to pass keyword arguments.

For example:
```
def func_that_takes_a_lot_of_params(some_required_param, **ctx):
    """
    Info that includes what is expected to be in context
    """
    self.some_required_param = some_required_param

    # perform any param checking early and raise if something is wrong
    if "something_important" not in ctx:
        raise ValueError("Something important is missing!")

    # store context 
    self.context = dict()

    for k,v in ctx.items():
        # perform any param-specific initialization here
        if k == "param_that_needs_special_handling":
            do_something_special(k)

        # keep a reference to the context params in self.context
        self.context[k] = v
```

This allows caller to use the function in 2 ways:

```
context = dict(param1=1, param2=2, param3=3)
func_that_takes_a_lot_of_params(some_param, **context)
```
or
```
func_that_takes_a_lot_of_params(some_param, param1=1, param2=2, param3=3)
```

Reply:
Many thanks for this tip @Drulex !! great great
I have opened an issue [here](https://github.com/neuropoly/ivado-medical-imaging/issues/275).
I would be happy to implement it in a dedicated PR